File: P3/QytetetJava/src/modeloQytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloQytetet;

/**
 *
 * @author marquirj
 */
public class Casilla {
    private int numCasilla;
    private int coste;
    private TipoCasilla tipo;
    private TituloPropiedad titulo;
    
    Casilla(int numcasilla, int coste, TipoCasilla tipo){
        this.numCasilla=numcasilla;
        this.coste=coste;
        this.tipo=tipo;
        this.titulo=null;
    }
    Casilla(int numCasilla,TipoCasilla tipo, TituloPropiedad titulo){
        this.numCasilla=numCasilla;
        this.coste=titulo.getPrecioCompra();
        this.tipo=tipo.CALLE;
        setTitulo(titulo);
    }

    public int getNumeroCasilla() {
        return numCasilla;
    }

    public int getCoste() {
        return coste;
    }

    public TipoCasilla getTipo() {
        return tipo;
    }

    public TituloPropiedad getTitulo() {
        return titulo;
    }
    private void setTitulo(TituloPropiedad tit){
        this.titulo=tit;
    }
    int pagarAlquiler(){
       int costeAlquiler=titulo.calcularImporteAlquiler();
       propietario.modificarSaldo(costeAlquiler);
    }
    boolean propietarioEncarcelado(){
        return this.titulo.propietarioEncarcelado();
    }
    boolean soyEdificable(){
        return this.tipo == tipo.CALLE;
    }
    boolean tengoPropietario(){
        return this.titulo.tengoPropietario();
    }
    TituloPropiedad asignarPropietario(Jugador jugador){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    @Override
    public String toString() {
        String aux;
        if(this.tipo!=tipo.CALLE){
            aux="No es una calle";
        }
        else
            aux=this.titulo.toString();
        
        return "Casilla{" + "numCasilla=" + numCasilla + ", coste=" + coste + ", tipo=" + tipo + ", titulo=" + aux + "}\n";
    }
    
}

File: P3/QytetetRuby/lib/casilla.rb
#ENCODING: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module QytetetRuby
class Casilla
  private_class_method :new
  attr_reader :numeroCasilla, :coste, :titulo, :tipo
  attr_writer :titulo
  private :titulo, :titulo=
  def initialize(numeroCasilla_,coste_,tipo_,titulo_)
    @numeroCasilla = numeroCasilla_
    @coste = coste_
    @tipo = tipo_
    @titulo = titulo_
  end
  def self.noTipoCalle(numeroCasilla_,coste_,tipo_)
    new(numeroCasilla_,coste_,tipo_,nil)
  end
  def self.tipoCalle(numeroCasilla_,tipo_,titulo_)
    new(numeroCasilla_,titulo_.precioCompra,tipo_,titulo_)
  end
  
  def asignarPropietario(jugador)
   @titulo=jugador
  end
  def pagarAlquiler
   costeAlquiler=@titulo.pagarAlquiler()
  end
  def propietarioEncarcelado
    return @titulo.propietarioEncarcelado
  end
  def soyEdificable
    return (@tipo == TipoCasilla::CALLE)
  end
  def tengoPropietario
    return @titulo.tengoPropietario
  end
  
  
  
  
  def to_s
    "Numero Casilla : #{@numeroCasilla}\n
     Coste: #{@coste} \n
     Tipo: #{@tipo}\n
     Titulo Propiedad: #{@titulo.to_s}\n
     "
  end
end
end